lib/rspec-puppet/facter_impl.rb

```diff
@@ -8,7 +8,11 @@ def initialize
     end
 
     def value(fact_name)
-      @facts[fact_name.to_s]
+      begin
```

Comment:
I'm so used to Ruby 2.5+ where you can write:
```ruby
def
  @facts.dig(*fact_name.to_s.split('.'))
rescue TypeError
  nil
end
```

But this gem doesn't declare to be Ruby 2.5+ so I think your current code is fine.